thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java

```diff
@@ -539,6 +549,22 @@ private String getDimensions(String dataset, String exploreDimensions) throws Ex
     return reorderedExploreDimensions.toString();
   }
 
+  // Activate anomaly function
+  @POST
+  @Path("/anomaly-function/activate")
+  public Response activateAnomalyFunction(@NotNull @QueryParam("functionId") Long id, @QueryParam("modifiedFunctionName") String functionName){
+    if (id == null) {
+      throw new IllegalArgumentException("id is a required query param");
```

Comment:
We should return "400 Bad Request" because the user wouldn't see the message of the exception on the frontend.